src/seedcase_sprout/core/read_properties.py

```diff
@@ -47,4 +51,8 @@ def read_properties(path: Path) -> PackageProperties:
     """
     check_is_file(path)
     properties = read_json(path)
+    if properties.get("resources"):
+        check_properties(properties)
+    else:
+        check_package_properties(properties)
```

Comment:
```suggestion
    check_properties(
        properties,
        ignore=[CheckErrorMatcher(validator="required", json_path=r"resources$")],
    )
```
Only because `properties.get` will throw if `properties` is valid JSON but not a `dict`.
So it needs some sort of `isinstance(properties, dict)` check. I will add this check in `check_properties` because it fits nicely in there and then the error thrown will be a nice `CheckError`.

The ignore bit here ignores missing resources (resources = None).
So far, we haven't considered empty resources (resources = []) correct because that's not allowed in Frictionless. But you can ignore that error by adding `CheckErrorMatcher(validator="minItems", json_path=r"resources$")`.

Reply:
I knew you had done a better way, that's much better, thanks!!